### PR TITLE
Add PreviousSet and PreviousClear functions

### LINKED ISSUE #86: get top bit set?
Is there no way to get the top bit but to iterate from set.Len()-1 down and set.Test?

### PR DESCRIPTION
**Pull request by lukesandberg**
Add PreviousSet and PreviousClear functions

These are analogous to NextSet and NextClear but probing in the opposite direction.

Fixes #86

### CODE REVIEW

File: bitset.go [hunk 1]
```diff
@@ -586,6 +586,68 @@ func (b *BitSet) NextClear(i uint) (uint, bool) {
...
+	// Flip all bits and find the highest one bit
+	w = ^w
+	// Clear the bits above the index
+	w = w & ((1 << (wordsIndex(i) + 1)) - 1)
+	if w != 0 {
+		return uint(x<<log2WordSize) + len64(w) - 1, true
+	}
+
+	x--
+	// bounds check elimination in the loop
```
Comment (lemire) [inline bitset.go:636]: Does this optimization works?

File: bitset.go [hunk 2]
```diff
@@ -586,6 +586,68 @@ func (b *BitSet) NextClear(i uint) (uint, bool) {
...
+	}
+	w := b.set[x]
+	// Clear the bits above the index
+	w = w & ((1 << (wordsIndex(i) + 1)) - 1)
+	if w != 0 {
+		return uint(x<<log2WordSize) + len64(w) - 1, true
+	}
+	x--
+	// bounds check elimination in the loop
+	if x < 0 {
```
Comment (lemire) [inline bitset.go:605]: Does this work?

**Comment by lemire** [comment]
@lukesandberg How did you verify that the bound check elimination trick worked?

**Comment by lemire** [comment]
Running tests.

File: bitset.go (hunk 1 above)
Comment (lukesandberg) [reply bitset.go]: It does not.  Dropped and simplified the following loop

File: bitset.go (hunk 2 above)
Comment (lukesandberg) [reply bitset.go]: ditto

**Comment by lukesandberg** [comment]
Removed the bounds check elimination since it was non functional.

**Comment by lemire** [comment]
Running tests. We shall merge soon after they go green.

**Comment by lukesandberg** [comment]
ping

**Pull request merged**